qiskit_experiments/calibration_management/calibrations.py

```diff
@@ -213,22 +237,34 @@ def get_template(
             CalibrationError: if no template schedule for the given schedule name and qubits
                 was registered.
         """
-        key = ScheduleKey(schedule_name, self._to_tuple(qubits))
+        n_qubits = self._get_n_qubits(schedule_name)
 
-        if key in self._schedules:
-            return self._schedules[key]
-
-        if ScheduleKey(schedule_name, ()) in self._schedules:
-            return self._schedules[ScheduleKey(schedule_name, ())]
+        for qubit_tuple in [self._to_tuple(qubits), ()]:
+            key = ScheduleKey(schedule_name, qubit_tuple, n_qubits)
+            if key in self._schedules:
+                return self._schedules[key]
 
         if qubits:
             msg = f"Could not find schedule {schedule_name} on qubits {qubits}."
         else:
-            msg = f"Could not find schedule {schedule_name}."
+            msg = f"Could not find schedule {schedule_name} on {n_qubits} qubits."
 
         raise CalibrationError(msg)
 
-    def remove_schedule(self, schedule: ScheduleBlock, qubits: Union[int, Tuple[int, ...]] = None):
+    def _get_n_qubits(self, schedule_name: str) -> Tuple:
+        """Return the number of qubits for this schedule."""
+        for key in self._schedules:
+            if key.schedule == schedule_name:
```

Comment:
What happens if we have two schedules with the same name but with different number of qubits? Is there any logic that prevents storing them?

Reply:
Yes we have such a logic. See here: https://github.com/Qiskit/qiskit-experiments/blob/3d2f8e94e2ee19d63715b275754289e0d11acdd7/qiskit_experiments/calibration_management/calibrations.py#L133-L141